pkg/log/grpc-logger.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019-2020 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"fmt"
+	"google.golang.org/grpc/grpclog"
+)
+
+// SetGrpcLogger sets up a logger for (google.golang.org/)grpc.
+func SetGrpcLogger(source string, rate *Rate) {
+	var l Logger
+
+	if source == "" {
+		l = Default()
+	} else {
+		l = log.get(source)
+	}
+
+	if rate != nil {
+		l = RateLimit(l, *rate)
+	}
+
+	grpclog.SetLoggerV2(&grpclogger{Logger: l})
+}
+
+// grpclogger implements grpclog.LoggerV2 interface for our logger.
+type grpclogger struct {
+	Logger
+}
+
+func (g grpclogger) Info(args ...interface{}) {
+	g.Logger.Debug("%s", fmt.Sprint(args...))
+}
+
+func (g grpclogger) Infoln(args ...interface{}) {
+	g.Logger.Debug("%s", fmt.Sprint(args...))
+}
+
+func (g grpclogger) Infof(format string, args ...interface{}) {
+	g.Logger.Debug("%s", fmt.Sprintf(format, args...))
+}
```

Comment:
It's more the nature/semantics than the volume of messages. Anything below warnings from grpc really looks like debug messages to me. Since debugging messages we can still control individually per logger source, if you need to see the ones from gRPC, you can do so without necessarily/potentially getting flooded by a ton of others.

Reply:
Sounds reasonable, thanks